Squeezer/src/main/java/uk/org/ngo/squeezer/itemlist/JiveItemViewLogic.java

```diff
@@ -157,13 +157,20 @@ private void showContextMenu(final ViewParamItemView<JiveItem> viewHolder, final
         if (preferences.isDownloadEnabled() && contextMenuItem != null && contextMenuItem.canDownload()) {
             menu.add(Menu.NONE, index++, Menu.NONE, R.string.DOWNLOAD);
         }
+        if (contextMenuItem != null) {
+            menu.add(Menu.NONE, index++, Menu.NONE, "Play random"); // TODO Context menu 1
+        }
+
         final int offset = index;
         for (JiveItem jiveItem : items) {
             menu.add(Menu.NONE, index++, Menu.NONE, jiveItem.getName()).setEnabled(jiveItem.goAction != null);
         }
 
         contextPopup.setOnMenuItemClickListener(menuItem -> {
-            if (menuItem.getItemId() < offset) {
+            if (menuItem.getItemId() == offset - 1) {
```

Comment:
This is wrong.
If item is not a music folder (contextMenuItem.moreAction.action.cmd.contains("folderinfo") then offset - 1 is the download menu entry.

Reply:
Fixed it by also checking for "folderinfo".

Reply:
This may still be the download entry if "contextMenuItem.randomPlayFolderCommand() == null". You can improve the readability and correct this by extracting the test into a function e.g. canRandomPlayFolder(contextMenuItem).

Reply:
Thanks, fixed it. Found a bug.